src/EditorFeatures/Core/SymbolSearch/SymbolSearchUpdateEngineFactory.cs

```diff
@@ -38,144 +41,63 @@ private class RemoteUpdateEngine : ISymbolSearchUpdateEngine
             private readonly SemaphoreSlim _gate = new SemaphoreSlim(initialCount: 1);
 
             private readonly Workspace _workspace;
-            private readonly ISymbolSearchLogService _logService;
-            private readonly CancellationToken _cancellationToken;
-
-            private RemoteHostClient _client;
-            private RemoteHostClient.Session _sessionDoNotAccessDirectly;
+            private readonly KeepAliveSession _session;
```

Comment:
What does hte name "KeepAliveSession" supposed to indicate to me?  How would it be different from just a "RemoteSession"?